### ISSUE TITLE
Default map location 

### ISSUE THREAD

**Issue by alensiljak** [ev-map/EVMap#191]
I would like to be able to set the starting map location instead of having to zoom in from space to a location I normally look at.
A convenient option might be a menu item "save current view as default" or something along those lines.

**Comment by MisterY** [comment]
Most likely. I keep the location turned off since it is not possible to have only triangulation and having GPS on wastes a lot of battery.

**Comment by johan12345** [comment]
Only triangulation is also possible, but only starting with Android 12 where you can choose this during the permission request.

So yeah, it is a valid (albeit less common) use case. I guess one would want an option in settings to choose between
- Default (~Europe)
- Zoom to current location (not available if location permission disabled) 
- Custom view (which then allows you to choose on the map)

**Comment by MartinH-open** [comment]
What about a default view that displays the area of all your favorite locations - if you have defined some.
Else default to all charging locations (eg. Europe).
First choice: Zoom to current location (not available if location permission disabled)

**Comment by johan12345** [comment]
I just checked what Google Maps does when there is no current location available (or permission disabled): It simply keeps the position you viewed when the app was open the last time. I guess this is a pretty convenient solution that is easy to implement and doesn't require an extra menu/settings option.

Using the favorites might also be useful, but this depends on what you use favorites for (e.g. if you have some chargers as favorites around your hometown, but also some at a far away vacation destination, having a default view that shows all favorites is not that much better than showing all of Europe).

**Comment by johan12345** [comment]
With 0d11e45 I implemented the behavior explained above (keep the last viewed location if location data are not available).
With 71c36fb I adjusted it again, so that it will also keep the last viewed location if location data *are* available, but the user wasn't actually looking at their location the last time they used the app. This is now really equivalent to how e.g. Google Maps does it, and should in my opinion be good enough for most use-cases.
